File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> frequency = new HashMap<>();
        PriorityQueue<Integer> pq = new PriorityQueue<>((a,b) -> frequency.get(a) - frequency.get(b));

        for (int i = 0; i<nums.length; i++) {
            frequency.put(nums[i], frequency.getOrDefault(nums[i], 0) + 1);
        }

        final int copyK = k;
        frequency.entrySet().stream().forEach(entry -> {
            pq.offer(entry.getKey());

            if (pq.size() > copyK) {
                pq.poll();
            }
        });

        List<Integer> result = new ArrayList<>();
        while(!pq.isEmpty() && k > 0) {
            result.add(pq.poll());
            k--;
        }

        return result.stream().mapToInt(i -> i).toArray();
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        chars = {}
        result = 0
        i, j = 0, 0
        while j<len(s):
            if s[j] in chars:
                i = max(chars[s[j]] + 1, i)
            chars[s[j]] = j
            result = max(result, j - i + 1)
            j += 1
        return result